app/models/concerns/attribute_blockable.rb

```diff
@@ -0,0 +1,372 @@
+# frozen_string_literal: true
+
+# AttributeBlockable provides a flexible system for checking if model attributes
+# are blocked via the BlockedObject system, with support for efficient N+1 query
+# prevention through eager loading.
+#
+# == Basic Usage
+#
+# To make an attribute blockable, use the +attr_blockable+ class method:
+#
+#   class User < ApplicationRecord
+#     include AttributeBlockable
+#
+#     attr_blockable :email
+#     attr_blockable :form_email_domain, object_type: :email
+#   end
+#
+# This generates several instance methods:
+# - +blocked_by_email?+ - Returns true if the email is blocked
+# - +blocked_by_email_at+ - Returns the timestamp when blocked, or nil
+# - +block_by_email!+ - Blocks the email value
+# - +unblock_by_email!+ - Unblocks the email value
+# - +blocked_emails+ - Returns array of blocked email values
+# - +blocked_emails_objects+ - Returns BlockedObject records
+#
+# == Preventing N+1 Queries
+#
+# When loading multiple records that need blocked attribute checks, use
+# +with_blocked_attributes_for+ to preload blocked status for all records
+# in a single MongoDB query:
+#
+#   # Without preloading (N+1 queries)
+#   users = User.where(verified: true)
+#   users.each { |user| puts user.blocked_by_email? }
+#
+#   # With preloading (single query)
+#   users = User.where(verified: true)
+#               .with_blocked_attributes_for(:email, :email_domain)
+#   users.each { |user| puts user.blocked_by_email? }
+#
+# == Caching
+#
+# The concern uses an instance variable to cache blocked status. This cache is
+# automatically populated when using +with_blocked_attributes_for+ or lazily
+# loaded on first access. The cache is cleared on +reload+.
+#
+# == Introspection
+#
+# The concern tracks all blockable attributes defined on a model:
+#
+#   User.blockable_attributes
+#   # => [{ attribute: :email, blockable_method: :email },
+#   #     { attribute: :email, blockable_method: :form_email }]
```

Comment:
Potentially wrong key in the attributes hash?

```ruby
[1] pry(main)> User.blockable_attributes
=> [{object_type: :email, blockable_method: :email},
 {object_type: :email, blockable_method: :form_email},
 {object_type: :email_domain, blockable_method: :email_domain},
 {object_type: :email_domain, blockable_method: :form_email_domain},
 {object_type: :ip_address, blockable_method: :account_created_ip}]
```

Reply:
I slipped when updating the documentation when changing `:attribute` to `:object_type` to better reflect what it represents, following Jono's suggestion. Thanks for catching it, will update.